x-pack/plugins/security/server/config.ts

```diff
@@ -332,35 +336,66 @@ export function createConfig(
                   ? { enabled: true, showInSelector: true, order, ...config.authc[providerType] }
                   : { enabled: true, showInSelector: true, order },
             };
+
             return legacyProviders;
           },
           {} as Record<string, unknown>
         )
       : config.authc.providers
   ) as ProvidersConfigType;
 
+  const updatedProvidersWithAccessAgreement: Record<string, object> = {};
+
   // Remove disabled providers and sort the rest.
   const sortedProviders: Array<{
     type: keyof ProvidersConfigType;
     name: string;
     order: number;
     hasAccessAgreement: boolean;
   }> = [];
+
   for (const [type, providerGroup] of Object.entries(providers)) {
     for (const [name, { enabled, order, accessAgreement }] of Object.entries(providerGroup ?? {})) {
+      const hasLocalAccessAgreement: boolean = !!accessAgreement?.message;
```

Comment:
__question:__ wouldn't it be easier (implementation/readability) if we keep global and provider-specific `accessAgreement` configs separate and just slightly update `Authenticator.shouldRedirectToAccessAgreement` and `/security/access_agreement` route handler to account for the global access agreement message if provider doesn't have any? It'd also allow us to diverge these configs in the future if we want to (e.g. we can restrict global access agreement settings that providers can or cannot override, like we did for `xpack.session.cleanupInterval`).

What do you think?

Reply:
Yea, good call! Ill make the change